crates/y-sweet-worker/src/s3_store.rs

```diff
@@ -0,0 +1,133 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use reqwest::{Client, StatusCode};
+use rusty_s3::{Bucket, Credentials, S3Action};
+use std::{cell::OnceCell, time::Duration};
+use time::OffsetDateTime;
+use y_sweet_core::store::Store;
+
+const PRESIGNED_URL_DURATION_SECONDS: u64 = 60 * 60;
+pub struct S3Store {
+    bucket: Bucket,
+    init_bucket: OnceCell<()>,
+    credentials: Credentials,
+    prefix: Option<String>,
+    presigned_url_duration: Duration,
+}
+
+impl S3Store {
+    pub fn new(
+        region: String,
+        bucket_name: String,
+        prefix: Option<String>,
+        aws_access_key_id: String,
+        aws_secret: String,
+    ) -> Self {
+        let credentials = Credentials::new(aws_access_key_id, aws_secret);
+        let endpoint = format!("https://s3.dualstack.{}.amazonaws.com", region)
```

Comment:
not required for this PR, but we will want to make this configurable as well to support other S3-compatible stores

Reply:
yep, was actually thinking of doing it with minio for testing purposes